media/jsonp/common/src/main/java/io/helidon/media/jsonp/common/JsonpBodyWriter.java

```diff
@@ -50,15 +49,14 @@ public boolean accept(GenericType<?> type, MessageBodyWriterContext context) {
 
     @Override
     public Publisher<DataChunk> write(Single<JsonStructure> content, GenericType<? extends JsonStructure> type,
-            MessageBodyWriterContext context) {
+                                      MessageBodyWriterContext context) {
 
         MediaType contentType = context.findAccepted(MediaType.JSON_PREDICATE, MediaType.APPLICATION_JSON);
         context.contentType(contentType);
-        return content.flatMap(new JsonStructureToChunks(jsonWriterFactory, context.charset()));
+        return content.map(new JsonStructureToChunks(jsonWriterFactory, context.charset()));
```

Comment:
I under what you are doing here, but if we do this we need to do this for all the places where there is similar code. i.e  jsonb, jackson.

Reply:
this is not a different functionality - it still works the same, I am just skipping going from an immediate value to publisher, as that created an unnecessary `flatMap` operation.
Also when I need to process records one by one, the intermediate publisher would cause trouble (as if it had more than one chunk per `JsonObject`, I would insert two commas there).

We should revisit it when implementing the stream writers for `JSON-B` and for `Jackson`

Reply:
will add a reference to those issues